_posts/2021-06-21-a-new-tale.md

```diff
@@ -0,0 +1,43 @@
+---
+title: A new Tale - Pulp gets queueless distributed task workers
+author: Matthias Dellweg
+tags:
+  - tasking
+  - performance
+  - high availability
+  - scalability
+---
+
+Pulp is about to introduce a new replacement for its tasking system.
+This complete redesign comes with a much simpler architecture that will drop the need for the resource-manager, the last singleton in the Pulp 3 infrastructure.
+Additionally, the distributed nature of the new task worker type drops the use of `Redis Queue` (`rq`).
+With this simplification there is no longer a need to replicate task data accross two databases.
+The main Postgres Pulp database will therefore be the single source of truth about tasks.
+Last but not least, the new worker type will use Postgres advisory locks to ensure exclusive access to tasks.
+This is a much more lightweight approach than storing access locks in a separate table.
+Also it simplifies the cleanup process vastly if a worker dies as postgres will clean the locks with its connection.
+
+With all these simplifications, the new tasking system is more reliable.
+Without the resource-manager as the single point of failure, it will be fully high available.
+Also with the resource-manager seen as the choke point in task processing, the new distributed system will scale much better in performance with the number of parallel running workers.
+
+> ![Tasking system benchmark](/images/tasking_system_benchmark/scale_workers.png) <br/>
```

Comment:
This image doesn't show up for me.

Reply:
Same here, but i tried to follow the examples in `git grep "/images"`:
```
...
_posts/2020-11-25-pulp-community-survey-results-2020.md:![Interested in contributing to Pulp?](/images/communitysurvey/2020/interest-in-contributing.png)
```

Reply:
Strange, this worked here: https://github.com/pulp/pulpproject.org/pull/317/files#diff-4ebb213dffb16eafe554603b7b4b80a837e9e65e47deeb23e68562e63be0ea4dR56 also

Reply:
@mcorr do you _know_ what's the right approach here?

Reply:
I'll take a look now @mdellweg !

Reply:
Serving this locally and it is working for me @mdellweg